File: OBJCSample/OBJCSample/viewControllers/TodayViewController.m
//
//  TodayViewController.m
//  ObjSample
//
//  Created by Ali on 10/23/16.
//  Copyright © 2016 Ali Azadeh. All rights reserved.
//

#import "TodayViewController.h"
@import AZCalendarOBJC;
@interface TodayViewController ()

@end

@implementation TodayViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setDate];
    [self setDayOfWeek];
    [self setMonthName];
   
    
    
    
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/



- (void) setDate{

    

    
    DateType *todayGregorian = [AZCalendar getCurrentGregorianDate];
    DateType *todayPersian = [AZCalendar getCurrentPersianDate];
    DateType *todayHijri = [AZCalendar getCurrentHijriDate];
    
    switch ([_segCtrCalendarType selectedSegmentIndex]) {
        case 0:
            // Georgian
            _lblDay.text = [NSString stringWithFormat:@"%ld",(long)todayGregorian.day];
             _lblMonth.text = [NSString stringWithFormat:@"%ld",(long)todayGregorian.month];
             _lblYear.text = [NSString stringWithFormat:@"%ld",(long)todayGregorian.year];
            break;
            
        case 1 :
            // Persian
            _lblDay.text = [NSString stringWithFormat:@"%ld",(long)todayPersian.day];
            _lblMonth.text = [NSString stringWithFormat:@"%ld",(long)todayPersian.month];
            _lblYear.text = [NSString stringWithFormat:@"%ld",(long)todayPersian.year];
            break;
            
            case 2:
            // Hijri or Islamic
            _lblDay.text = [NSString stringWithFormat:@"%ld",(long)todayHijri.day];
             _lblMonth.text = [NSString stringWithFormat:@"%ld",(long)todayHijri.month];
             _lblYear.text = [NSString stringWithFormat:@"%ld",(long)todayHijri.year];
            break;

    }
  
}

-(void)setDayOfWeek{

    self.lblDayOfWeek.text = [AZCalendar getDayNameWithCalendar:gregorian date:[AZCalendar getCurrentGregorianDate]];
    
}

-(void)setMonthName{
    
    switch ([_segCtrCalendarType selectedSegmentIndex]) {
            
            case 0:
            _lblMonthName.text = [AZCalendar getMonthNameWithCalendar:gregorian date:[AZCalendar getCurrentGregorianDate]];
            break;
            case 1:
            _lblMonthName.text = [AZCalendar getMonthNameWithCalendar:persian date:[AZCalendar getCurrentPersianDate]];
            break;
            case 2:
            _lblMonthName.text = [AZCalendar getMonthNameWithCalendar:hijri date:[AZCalendar getCurrentHijriDate]];
            break;
            default:
            break;
            
    }

}




- (IBAction)changedCalendarMode:(id)sender {
    [self setDate];
    [self setDayOfWeek];
    [self setMonthName];
   
    
}
@end

File: SwiftSample/SwiftSample/ViewControllers/TodayViewController.swift
//
//  TodayViewController.swift
//  SwiftSample
//
//  Created by Ali on 10/26/16.
//  Copyright © 2016-2024 Ali Azadeh. All rights reserved.
//

import UIKit

final class TodayViewController: UIViewController {
    
    // MARK: - IBOutlets
    @IBOutlet private weak var segCtrCalendarType: UISegmentedControl!
    @IBOutlet private weak var lblDay: UILabel!
    @IBOutlet private weak var lblMonth: UILabel!
    @IBOutlet private weak var lblYear: UILabel!
    @IBOutlet private weak var lblMonthName: UILabel!
    @IBOutlet private weak var lblDayOfWeek: UILabel!
    
    // MARK: - Lifecycle
    override func viewDidLoad() {
        super.viewDidLoad()
        updateDisplay()
    }
    
    // MARK: - IBActions
    @IBAction private func changedCalendarMode(_ sender: UISegmentedControl) {
        updateDisplay()
    }
    
    // MARK: - Private Methods
    private func updateDisplay() {
        setDate()
        setDayOfWeek()
        setMonthName()
    }
    
    private func setDate() {
        let selectedIndex = segCtrCalendarType.selectedSegmentIndex
        
        let date: DateType
        switch selectedIndex {
        case 0:
            date = AZCalendar.getCurrentGregorianDate()
        case 1:
            date = AZCalendar.getCurrentPersianDate()
        case 2:
            date = AZCalendar.getCurrentHijriDate()
        default:
            date = AZCalendar.getCurrentGregorianDate()
        }
        
        lblDay.text = "\(date.day)"
        lblMonth.text = "\(date.month)"
        lblYear.text = "\(date.year)"
    }
    
    private func setDayOfWeek() {
        let gregorianDate = AZCalendar.getCurrentGregorianDate()
        lblDayOfWeek.text = AZCalendar.getDayName(calendar: .gregorian, date: gregorianDate)
    }
    
    private func setMonthName() {
        let selectedIndex = segCtrCalendarType.selectedSegmentIndex
        
        let monthName: String
        switch selectedIndex {
        case 0:
            monthName = AZCalendar.getMonthName(calendar: .gregorian, date: AZCalendar.getCurrentGregorianDate())
        case 1:
            monthName = AZCalendar.getMonthName(calendar: .persian, date: AZCalendar.getCurrentPersianDate())
        case 2:
            monthName = AZCalendar.getMonthName(calendar: .hijri, date: AZCalendar.getCurrentHijriDate())
        default:
            monthName = ""
        }
        
        lblMonthName.text = monthName
    }
}
